mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -525,6 +558,15 @@ func fromOnlineArchiveToMap(in *admin.BackupOnlineArchive) map[string]any {
 		schemaVals["data_expiration_rule"] = []any{dataExpirationRule}
 	}
 
+	var dataProcessRegion map[string]any
+	if in.DataProcessRegion != nil && in.DataProcessRegion.CloudProvider != nil && in.DataProcessRegion.Region != nil {
```

Comment:
since both field are optional, would it be possible only one can be defined?
Even though docs shows both fields as optional, it would seem logical that both cloud provider and regions must be defined.

Reply:
I would leave it like that. `Optional` gives flexibility for future changes as it leaves the schema "free". At the same time it's obvious that these two fields go together: if you specify the region you'll want to specify the provider. If you specify the provider, you _might_ want to specify the region.

Reply:
I usually like going with a more restrictive schema for new attributes for the following reason: In the future you will always be able to change from required to optional if new attributes are defined or a new use case appears, however going from optional to required would be a breaking change. Not blocking the PR but just to keep in mind.

Reply:
we haven't decided about where to do checks (client-side or server-side) but I tend to prefer to do changes in the server so logic is centralised.  

At the moment I don't see a big problem if an attribute is defined as optional in TF but at some moment server complains that it's required if not provided.

Reply:
Agustin, I've expressed myself wrongly and mixed up two concepts. My sentence makes sense replacing `Optional` with `Required`, which is what you've said. Sorry about that.

In general, in fact, I think I support the idea that we should minimise validation in our provider because a) makes our code easier b) reduces code duplication c) keeps it consistent with API behaviour.

Said that, by having read this again I think we **need to fix this logic**.

Reply:
```suggestion
	if in.DataProcessRegion != nil && (in.DataProcessRegion.CloudProvider != nil || in.DataProcessRegion.Region != nil) {
```

@AgustinBettati @lantoli  wdyt about this change? I think it should be enough because if one of the two fields is nil we would pass it to the API as nil which shouldn't introduce any weird behaviour and additionally should return client errors from the API if needed. 

Open to other suggestions

Reply:
Yes agree with your suggested change given that both fields are defined as optional.